staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -76,15 +82,35 @@ func WriteObject(statusCode int, gv schema.GroupVersion, s runtime.NegotiatedSer
 	io.Copy(writer, out)
 }
 
-// WriteObjectNegotiated renders an object in the content type negotiated by the client
-func WriteObjectNegotiated(s runtime.NegotiatedSerializer, gv schema.GroupVersion, w http.ResponseWriter, req *http.Request, statusCode int, object runtime.Object) {
+// WriteObjectNegotiated renders an object in the content type negotiated by the client.
+// The context is optional and can be nil.
+func WriteObjectNegotiated(ctx request.Context, s runtime.NegotiatedSerializer, gv schema.GroupVersion, w http.ResponseWriter, req *http.Request, statusCode int, object runtime.Object) {
 	serializer, err := negotiation.NegotiateOutputSerializer(req, s)
 	if err != nil {
 		status := apiStatus(err)
 		WriteRawJSON(int(status.Code), status, w)
 		return
 	}
 
+	// complete audit event in the context
+	if ctx != nil {
+		ae := request.AuditEventFrom(ctx)
+		if ae != nil && ae.Level.GreaterOrEqual(audit.LevelResponse) {
+			if status, ok := object.(*metav1.Status); ok {
+				ae.ResponseStatus = status
```

Comment:
it's cheap, so why not. And ResponseStatus looks very meta-like, only technically we need to go down to the rest handler level to get that value.